### ISSUE TITLE
Change api url after app loading

### ISSUE THREAD

**Issue by zeevkatz** [reyesoft/ngx-jsonapi#35]
There is a way to change the api path after loading? I'm trying to load the api path from configuration JSON file for dynamic config options in production and set the url only after the config file loading.

**Comment by pablorsk** [comment]
hi @zeevkatz,

yeap! you can do with any service. For example:

@Injectable()
export class AuthorsService extends Service<Author> {
    public resource = Author;
    public type = 'authors';
    public schema: ISchema = {
        relationships: {
            photos: {
                hasMany: true
            }
        }
    };

    public getPath(): string {
        return 'my_particular_path_for_this_resource_type';
    }
}

**Comment by zeevkatz** [comment]
Maybe i was not clear, i'm trying to change the api url and not a specific service path (the url that defined in the module import).

**Comment by pablorsk** [comment]
Oh! ok....

I'm thinking on a feature... We can atach new method on service for specials URIs, o just a particular service base url... :thinking:

**Comment by zeevkatz** [comment]
Created new pull request with solution for this issue.
Waiting for your merge...

**Comment by zeevkatz** [comment]
@pablorsk When will you merge the pull request?

**Comment by zeevkatz** [comment]
@pablorsk Something new about the PR?

**Comment by zeevkatz** [comment]
Hi @pablorsk,
Please let me know if there are issues with my last pull request that related to this issue.
Jan 30 was really far from now and this is a really important change,
Thanks.

**Comment by valentasm1** [comment]
I am not sure if i am not too late for possible solution.  But maybe it will help for someone one day. Use case is to have one api url for deployment and other for prod. Also not 100% clear that it gonna work for all but you'll get idea.

Create folder under root "environments" with two files 
environment.ts
`export const environment = {
  production: false,
  someEndpoint : 'http://localhost:55180/',
  anotherEndPoint : "http://localhost:55009/x-files/"
};`
environment.prod.ts
`export const environment = {
  production: true,
  someEndpoint : 'http://api.superdomain.com/',
  anotherEndPoint : "http://api.superdomain.com/no-files/"
};`

And then in you service you could use
`private rootUrl = environment.someEndpoint + "x-files-documents/";
public getPath(): string {
	return this.rootUrl;
}`

So when you build with flag prod it gonna take variables from prod file. Easy peasy.

**Comment by pablorsk** [comment]
In [Multinexo](https://multinexo.com/), @valentasm1, we have a very similar code.
